tee-worker/core/rpc-client/src/ws_client.rs

```diff
@@ -82,37 +86,43 @@ impl WsClient {
 	pub fn connect_watch_with_control(
 		url: &str,
 		request: &str,
-		result: &MpscSender<String>,
+		result: MpscSender<String>,
 		control: Arc<WsClientControl>,
-	) -> Result<()> {
-		debug!("Connecting web-socket connection with watch");
-		connect(url.to_string(), |out| {
-			control.subscribe_sender(out.clone()).expect("Failed sender subscription");
-			WsClient::new(out, request.to_string(), result.clone(), true)
-		})
+	) {
+		debug!("Connecting web-socket connection with watch to server {url}");
+		loop {
```

Comment:
I get the idea, but I think we shouldn't put infinite loop in a primitive function of client API. If we need to reconnect, we should handle it in the caller scope.

Btw this function is only used by `DirectClient` which sends requests to the worker-ws via CLI - actually I don't think we need automatic reconnection there